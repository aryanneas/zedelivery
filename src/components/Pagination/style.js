import styled from 'styled-components'

export const PaginationItem = styled.button.attrs(({ active }) => ({
  className: active ? 'active' : '',
}))`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 40px;
  background-color: #fff;
  border-color: #fff;
  padding: 10px 15px;
  color: #000;
  text-align: center;
  border-top: 1px solid #d3d3d3;
  border-bottom: 1px solid #d3d3d3;
  border-right: 1px solid #d3d3d3;
  cursor: pointer;
  &:first-child {
    border-bottom-left-radius: 4px;
    border-top-left-radius: 4px;
    border-left: 1px solid #d3d3d3;
  }
  &:last-child {
    border-bottom-right-radius: 4px;
    border-top-right-radius: 4px;
  }
  &.active,
  &:hover {
    color: #fff;
    background-color: #ffc500;
    border-color: #ffc500;
  }

  &:disabled {
    color: #939393;
    background-color: #d3d3d3;
    border: none;
    cursor: not-allowed;
  }
`
export const Grid = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
`

export const FlexComponent = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 10px 0;
`
export const ChildrenWrapper = styled.div`
  padding: 10px 0;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  height: 54vh;
  overflow-y: scroll;
`
