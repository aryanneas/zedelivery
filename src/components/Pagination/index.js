import React, { useState } from 'react'

import {
  FiChevronsLeft,
  FiChevronRight,
  FiChevronsRight,
  FiChevronLeft,
} from 'react-icons/fi'

import { PaginationItem, Grid, FlexComponent, ChildrenWrapper } from './style'

const showPaginationItems = (
  totalPages,
  currentPageNumber,
  setCurrentPageNumber
) => {
  return (
    <>
      <PaginationItem
        disabled={currentPageNumber <= 1}
        onClick={() => setCurrentPageNumber(1)}
      >
        <FiChevronsLeft />
      </PaginationItem>
      <PaginationItem
        disabled={currentPageNumber <= 1}
        onClick={() => setCurrentPageNumber(currentPageNumber - 1)}
      >
        <FiChevronLeft />
      </PaginationItem>
      {currentPageNumber !== 1 ? (
        <PaginationItem
          onClick={() => setCurrentPageNumber(currentPageNumber - 1)}
        >
          {currentPageNumber - 1}
        </PaginationItem>
      ) : null}
      <PaginationItem active>{currentPageNumber}</PaginationItem>
      {currentPageNumber < totalPages ? (
        <PaginationItem
          onClick={() => setCurrentPageNumber(currentPageNumber + 1)}
        >
          {currentPageNumber + 1}
        </PaginationItem>
      ) : null}
      <PaginationItem
        disabled={currentPageNumber >= totalPages}
        onClick={() => setCurrentPageNumber(currentPageNumber + 1)}
      >
        <FiChevronRight />
      </PaginationItem>
      <PaginationItem
        disabled={currentPageNumber >= totalPages}
        onClick={() => setCurrentPageNumber(totalPages)}
      >
        <FiChevronsRight />
      </PaginationItem>
    </>
  )
}

const Pagination = ({ children, itemsPerPage, arrayItems }) => {
  const [currentPageNumber, setCurrentPageNumber] = useState(1)
  const [totalPages] = useState(
    Math.ceil((arrayItems || children.length) / itemsPerPage)
  )

  const paginate = (items, currentPage) => {
    if (!items) return null
    const start = (currentPage - 1) * itemsPerPage
    const end = start + itemsPerPage
    return items.slice(start, end)
  }

  return (
    <Grid>
      <FlexComponent>
        {showPaginationItems(
          totalPages,
          currentPageNumber,
          setCurrentPageNumber
        )}
      </FlexComponent>
      <ChildrenWrapper>{paginate(children, currentPageNumber)}</ChildrenWrapper>
      <FlexComponent>
        {showPaginationItems(
          totalPages,
          currentPageNumber,
          setCurrentPageNumber
        )}
      </FlexComponent>
    </Grid>
  )
}

export default Pagination
