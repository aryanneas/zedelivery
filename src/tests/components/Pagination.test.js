import React from 'react'
import { render } from '@testing-library/react'

import Pagination from '../../components/Pagination'

const mock = [{ id: 1, name: 'test' }]

describe('Pagination component', () => {
  it('should render Pagination', async () => {
    const { asFragment } = render(
      <Pagination itemsPerPage={10} arrayItems={mock} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
