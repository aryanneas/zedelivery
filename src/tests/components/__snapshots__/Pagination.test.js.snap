// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render Pagination 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  background-color: #fff;
  border-color: #fff;
  padding: 10px 15px;
  color: #000;
  text-align: center;
  border-top: 1px solid #d3d3d3;
  border-bottom: 1px solid #d3d3d3;
  border-right: 1px solid #d3d3d3;
  cursor: pointer;
}

.c2:first-child {
  border-bottom-left-radius: 4px;
  border-top-left-radius: 4px;
  border-left: 1px solid #d3d3d3;
}

.c2:last-child {
  border-bottom-right-radius: 4px;
  border-top-right-radius: 4px;
}

.c2.active,
.c2:hover {
  color: #fff;
  background-color: #ffc500;
  border-color: #ffc500;
}

.c2:disabled {
  color: #939393;
  background-color: #d3d3d3;
  border: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 0;
}

.c3 {
  padding: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 54vh;
  overflow-y: scroll;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        disabled=""
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="11 17 6 12 11 7"
          />
          <polyline
            points="18 17 13 12 18 7"
          />
        </svg>
      </button>
      <button
        class="c2"
        disabled=""
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </button>
      <button
        class="c2 active"
      >
        1
      </button>
      <button
        class="c2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </button>
      <button
        class="c2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="13 17 18 12 13 7"
          />
          <polyline
            points="6 17 11 12 6 7"
          />
        </svg>
      </button>
    </div>
    <div
      class="c3"
    />
    <div
      class="c1"
    >
      <button
        class="c2"
        disabled=""
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="11 17 6 12 11 7"
          />
          <polyline
            points="18 17 13 12 18 7"
          />
        </svg>
      </button>
      <button
        class="c2"
        disabled=""
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </button>
      <button
        class="c2 active"
      >
        1
      </button>
      <button
        class="c2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </button>
      <button
        class="c2"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="13 17 18 12 13 7"
          />
          <polyline
            points="6 17 11 12 6 7"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
